jwt/audience.go

```diff
@@ -22,43 +21,25 @@ func NewAudience(u string) *Audience {
 // Validate performs error checking on the Audience's URL.
 func (aud *Audience) Validate() error {
 	rawAudience := string(*aud)
-	p := strings.SplitN(rawAudience, "/", 5)
-	if len(p) != 5 {
-		return errors.New("invalid audence length")
+	p := strings.SplitN(rawAudience, "/", 4)
+	if len(p) != 4 {
+		return errors.New("invalid audience length")
+	}
+
+	if p[0] != "" {
+		return errors.New("audience URL must start with slash")
 	}
 
 	if p[1] != "projects" {
 		return errors.New("audience must start with a project")
 	}
 
-	var (
-		projectNumber = p[2]
-		audienceType  = p[3]
-		audienceID    = p[4]
-	)
-
-	if _, err := strconv.ParseUint(projectNumber, 10, 64); err != nil {
-		return fmt.Errorf("%q is not a valid project number: %s", projectNumber, err.Error())
+	if projectNumber := p[2]; len(projectNumber) == 0 {
+		return fmt.Errorf("%q is not a valid project number: zero-length string", projectNumber)
```

Comment:
did you mean something like `"audience %q must have a non-empty project number", rawAudience` ? otherwise it will always print out `"" is not ...`